src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,18 +77,56 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.sendError(500, "An error occurred while fetching Youtube Posts");
       return;
     }
+
+    int currentPage = convertToInt(request.getParameter("currentPage"));
+    int postPerPage = convertToInt(request.getParameter("postPerPage"));
+
+    threadTitles.clear();
+    threadSentiments.clear();
+    threadLikes.clear();
+    threadUrls.clear();
     for (YoutubePost post : newPosts) {
       threadTitles.add(post.getTitle());
       threadSentiments.add(analyze.getSentimentScore(post.getContent()));
-      threadUpvotes.add(AnalyzedVideo.getRandomUpvote());
+      threadLikes.add(AnalyzedVideo.getRandomUpvote());
       threadUrls.add(post.getUrl());
+    }
 
-      threadInfoList.put(TITLE, threadTitles);
-      threadInfoList.put(SENTIMENT, threadSentiments);
-      threadInfoList.put(UPVOTES, threadUpvotes);
-      threadInfoList.put(URL, threadUrls);
+    int numOfPages = (int) Math.ceil(threadTitles.size() / postPerPage);
+    createCurrentPage(currentPage, postPerPage);
+    if (numOfPages == 0) {
+      numOfPages++;
     }
+    threadInfoList.put(NUMOFPAGES, numOfPages);
+    threadInfoList.put(TITLE, threadTitles);
+    threadInfoList.put(SENTIMENT, threadSentiments);
+    threadInfoList.put(LIKES, threadLikes);
+    threadInfoList.put(URL, threadUrls);
```

Comment:
What do you think of using a list of ThreadInfo? That would be simpler to understand and we would have a list of entities instead of multiple entities per thread